src/server/game/Maps/Map.cpp

```diff
@@ -679,10 +704,95 @@ void Map::VisitNearbyCellsOf(WorldObject* obj, TypeContainerVisitor<Trinity::Obj
             cell.SetNoCreate();
             Visit(cell, gridVisitor);
             Visit(cell, worldVisitor);
+
+            uint32 zoneId, areaId;
+            obj->GetZoneAndAreaId(zoneId, areaId);
+            // Handle possible respawns
+            RespawnCellAreaZone(cell.GetCellCoord().GetId(), zoneId, areaId);
```

Comment:
I'm sure this part doesn't work properly - each cell is visited exactly once (isCellMarked/markCell) but there can be more than 1 area/zone in a given cell so some areas/zones might never respawn depending on visiting order